release-process.rst

```diff
@@ -64,65 +80,144 @@ Example time line with only one major version at a time
+- Optimisations and internal ABI and API changes are also still allowed.
+
+Release Candidates
+------------------
+
+- Release Candidate 1
+
+	- Tag: ``$rd + 82`` (Sep 24, 2024)
+	- Release: ``$rd + 84`` (Sep 26, 2024)
+
+- Release Candidate 2: ``$rd + 98`` (Oct 10, 2024)
+- Release Candidate 3: ``$rd + 112`` (Oct 24, 2024)
+- Release Candidate 4: ``$rd + 126`` (Nov 07, 2024)
+
+More release candidates can be added on a two week cycle if necessary.
+
+With the first release candidate:
+
+- Internal API numbers need to be updated (``PHP_API_VERSION``,
+  ``ZEND_MODULE_API_NO``, and  ``ZEND_EXTENSION_API_NO``).
+- The release branch (``PHP-8.4``) needs to be created.
+
+After the first release candidate:
+
+- API and ABI changes are no longer allowed.
+- No new features, whether they are small, or require an RFC.
+
+
+General Availability
+--------------------
+
+- x.y.0 (8.4.0)
+
+	- Tag: ``$rd + 138`` (Nov 19, 2024)
+	- Release: ``$rd + 140`` (Nov 21, 2024)
+
+Released from the last Release Candidate tag, without changes between RC4 (or
+later) and GA.
+
+Bug Fix and Security Releases
+-----------------------------
+
+After general availability release:
 
-Timeline example for a release
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Until end of year 2 (for PHP 8.4: until Dec 31, 2026):
 
-- June
+	- A new release every 4 weeks, synchronised with other release branches.
+	- Bug fixes and security fixes.
 
-  - Decisions which features or changes will be in the next release
-  - 1st release alpha (may have many alpha)
+- Until end of year 3 (for PHP 8.4: until Dec 31, 2027):
 
-- At least one release per month, more at wish
-- September, RC phases, biweekly release
+	- Security fixes, and fixes to address regressions introduced during a
+	  normal bug fix release.
+	- Updates to ABI incompatible versions of dependent libraries on Windows.
+	- Release only when there is a security issue or regression issue to
+	  address
+	- Release when a otherwise normal bug fix release for other branches is
+	  also made. Exceptions can be made for high risk security issues or high
+	  profile regressions.
 
-  - each RC should go through the QA before being published
+- Until end of year 4 (for PHP 8.4: until Dec 31, 2028):
 
-    - usually 2 days
-    - running the various test suites (phpt, custom real life tests, platform specific tests). Some tests need a day to run
+	- Security fixes **only**
+	- Release only when there is a security issue.
+	- Release when a otherwise normal bug fix release for other branches is
+	  also made. Exceptions can be made for high risk security issues.
+	  profile regressions.
+	- Updates to ABI incompatible versions of dependent libraries on Windows
+	  are **not** done.
 
-- November, Final
 
-  - Last RC taken as final, golden release (no change between the last RC and the final version)
+*End of year here means:* The end of the year following the original planned release
+date of a GA release.
```

Comment:
As written, this statement indicates that "end of year" always means the end of the year following the GA release (i.e., if the GA release is in November of 2024, then the end of year is *always* December 31, 2024). I don't think that's what's intended, though.

```suggestion
*"End of year" means:* The end of the calendar year, i.e., Dec 31 at 23:59:59 UTC. The numbered years in the examples (e.g., "end of year 2") indicate the number of calendar years following the GA release. For example, if the GA release for PHP 8.4 occurs on Nov 21, 2024, then "end of year 2" is Dec 31, 2026 23:59:59 UTC.
```

Reply:
What I was trying to capture is that if the release would have slipped to January 5, 2025, the "end of year two" would be December 31, 2026, not **December 31, 2027**. So your phrasing doesn't help this. Mine has the "original planned release date" wording in it, and although perhaps not clear *enough*, it is what I meant to say. Yours does not. Try again? (Pretty please)

Reply:
How about:

```suggestion
*"End of year" means:* The end of the calendar year, i.e., Dec 31 at 24:00 UTC. The numbered years in the examples (e.g., "end of year 2") indicate the number of calendar years following the *original planned GA release date*. For example, if the planned GA release date for PHP 8.4 is Nov 21, 2024, then "end of year 2" is Dec 31, 2026, 24:00 UTC, even if the actual release date slips to Jan 9, 2025.
```